### ISSUE TITLE
Show error attempting to EJSON-ify objects with a circular references

### ISSUE THREAD

**Issue by timhaines** [meteor/meteor#1446]
tl;dr: Handle or give better error messages when an object with a circular reference is returned from a Meteor Method.

I had a really simple meteor method, that makes a call to aws's npm module.  I was seeing this error:

Internal exception while processing message { msg: 'method', method: 'compareObject', params: [], id: '4' }  undefined

Which with PR #1444 becomes:

Internal exception while processing message { msg: 'method', method: 'compareObject', params: [], id: '4' } Maximum call stack size exceeded undefined

After a lot of bug hunting I accidentally discovered that it's because the method I was calling was returning an object (I didn't expect it would - I just thought it would run the callback), and that value was being returned to the last line in my Meteor.method, which in turn was trying to return it to the client.

When I realize what's going on, console.log'ing the returned object shows { service: { client: [Circular], ...

So presumably Meteor is trying to serialize it to send to the client, and is getting stuck in the circular reference, causing the 'Maximum call stack size exceeded'.

**Comment by hwillson** [comment]
To help provide a more clear separation between feature requests and bugs, and to help clean up the feature request backlog, Meteor feature requests are now being managed under the https://github.com/meteor/meteor-feature-requests repository. 

This feature request will be closed here, but anyone interested in migrating this feature request to the new repository (to make sure it stays active), can click [here](https://github.com/meteor/meteor-feature-requests/issues/new?title=Show%20error%20attempting%20to%20EJSON-ify%20objects%20with%20a%20circular%20references&body=Migrated%20from:%20meteor/meteor%231446) to start the feature request migration process. This manual migration process is intended to help identify which of the older feature requests are still considered to be of value to the community. Thanks!
